src/usr/local/www/system_camanager.php

```diff
@@ -308,6 +317,7 @@
 
 		if (!$input_errors) {
 			write_config();
+			ca_trustedcerts_generate();
```

Comment:
Deleting the entire set of cert files and rewriting them provides no additional safety.

1) If the contained cert does not match the hash name then it will fail for the site.  The certs here are found for the site by their hash name.  They are not all read until a match is found.  That's why they are required to be hash named.

2) Symlinks are not being destroyed.  So anyone who wishes to tamper can do so simply by creating a link to a cert located anywhere on the system they want.

3) Anyone wishing to tamper can simply put a cert of their choosing in the CA bundle (/usr/local/share/certs/ca-root-nss.crt).  It is not purged and reloaded at boot either.

This argument of safety and trust is nonsensical.

No need to read the files in and compare.  Just create an array of file names config says should be there and add/remove accordingly.  Not much work at all really, and far less work than writing files that don't need to be written.  Deleting and rewriting the files is much more work for no benefit.

```
// Get list of trust files for CA certs from config.
foreach($config['ca'] as $ca) {
	$ca_hash = cert_get_hash($ca['crt']);
	$ca_certfiles[] = "{$g['etc_path']}/ssl/certs/{$ca_hash}.0";
}

// Remove cert trust files that have no corresponding CA cert in config.
foreach (glob("{$g['etc_path']}/ssl/certs/*") as $cert_file) {
	if (!in_array($cert_file, $ca_certfiles)) {
		unlink_if_exists($cert_file);
	}
}

// Add missing cert trust files for CA certs in config.  Remove cert trust files of CA certs in config that are not trusted or are revoked.
foreach ($a_ca as $id => $value) {
	ca_trustedcert_add($id);
}
```